solidity/contracts/ValidatorRegistration.sol

```diff
@@ -1,23 +1,113 @@
+// Adapted from: https://github.com/prysmaticlabs/prysm/blob/master/contracts/validator-registration-contract/validator_registration.sol
 pragma solidity ^0.4.23;
+pragma experimental ABIEncoderV2;
 
 contract ValidatorRegistration {
 
-  event Deposit(bytes32 _pubkey, uint256 _withdrawal_shard_id, address _withdrawal_address, bytes32 _randao_commitment);
+    event Eth1Deposit(
+        bytes previousReceiptRoot,
```

Comment:
should this be indexed? we will probably need to get this at a later time

Reply:
Depends on which arg(s) we'd like to index. Variably sized types aren't indexed as is in the log blooms: their hashes are indexed iirc.

Reply:
[This](https://ethereum.stackexchange.com/a/7170) might help, although I'm not sure if this behavior has been updated in solc or web3.

Reply:
I don't think it matters what order deposits were made. In which case this doesnt need to be indexed. Unless there's other performance benefits from it ?

Reply:
@aunyks @GregTheGreek okay cool, I think it should be fine then.

Reply:
Bringing this back from the dead. The spec outlines `initial_validator_deposits is a list of Deposit objects built according to the Eth1Deposit logs up to the deposit that triggered the ChainStart log, processed in the order in which they were emitted (oldest to newest)` 

Therefore i do believe we need to index these. Correct me if im worng?